### PR TITLE
Add missing property validationContext on error object

### PR DESCRIPTION
**Pull request by rubybb**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by climba03003 — CHANGES_REQUESTED** [review]
Can you add an unit test for the changes?

It should be added in this folder https://github.com/fastify/fastify/tree/main/test/types

File: fastify.d.ts [hunk 1]
```diff
@@ -149,6 +149,7 @@ type TrustProxyFunction = (address: string, hop: number) => boolean
 declare module 'fastify-error' {
   interface FastifyError {
     validation?: ValidationResult[];
+    validationContext?: string;
```
Comment (Eomm) [inline fastify.d.ts:152]: Is it not possible to define the values it can assume?

**Comment by rubybb** [comment]
> Can you add an unit test for the changes?
> 
> It should be added in this folder https://github.com/fastify/fastify/tree/main/test/types

The change is already tested by this line.
https://github.com/rubybb/fastify/blob/907c2e274bb9cc77695add824a7f285df2f1b87c/test/types/hooks.test-d.ts#L99

File: fastify.d.ts (hunk 1 above)
Comment (rubybb) [reply fastify.d.ts:152]: I think it's safer to just type as a string instead of a union.

**Comment by mcollina** [comment]
If it's already tested there is no need for this change. I _think_ it's a worthwhile change.

**Comment by climba03003** [comment]
> 
> 
> > Can you add an unit test for the changes?
> > It should be added in this folder https://github.com/fastify/fastify/tree/main/test/types
> 
> The change is already tested by this line. https://github.com/rubybb/fastify/blob/907c2e274bb9cc77695add824a7f285df2f1b87c/test/types/hooks.test-d.ts#L99
> 
> Also we can add the `hacktoberfest` tag to this repository? For more information: https://hacktoberfest.digitalocean.com/

The test should be something like
```ts
expectType<string | undefined>(error.validationContext)
```
because the test you mention will always pass for any changes for `FastifyError`.

**Comment by rubybb** [comment]
> because the test you mention will always pass for any changes in `FastifyError`.

There is no check for any other properties in the error object, why should this one be unique?

**Comment by climba03003** [comment]
> 
> 
> > because the test you mention will always pass for any changes in `FastifyError`.
> 
> There is no check for any other properties in the error object, why should this one be unique?

Because test should be exist to ensure to regression in future. The current test is not enough for this purpose.

**Comment by climba03003** [comment]
@rubybb Can you fix the missing import for the test? I think it is the last pieces before merge.

**Comment by Eomm** [comment]
Friendly ping @rubybb 

Would you like to push this PR over the line?

**Pull request closed without merging**